kola/tests/docker/docker.go

```diff
@@ -459,17 +470,75 @@ func dockerUserNoCaps(c cluster.TestCluster) {
 	}
 }
 
-// testDockerInfo test that docker info's output is as expected.  the expected
-// filesystem may be asserted as one of 'overlay', 'btrfs', 'devicemapper'
-// depending on how the machine was launched.
-func testDockerInfo(expectedFs string, c cluster.TestCluster) {
+// dockerContainerdRestart ensures containerd will restart if it dies. It tests that containerd is running,
+// kills it, the tests that it came back up.
+func dockerContainerdRestart(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	// verify containerd came up
+	if !testContainerdUp(c) {
+		c.Fatalf("Containerd did not start")
+	}
+
+	// kill it and verify it went down
+	pid, err := m.SSH("pgrep containerd")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if string(pid) == "" {
+		c.Fatalf("Could not find containerd pid")
+	}
+
+	if _, err = m.SSH("sudo kill " + string(pid)); err != nil {
+		c.Fatal(err)
+	}
+
+	if testContainerdUp(c) {
```

Comment:
This seems like it would be racy since systemd could have already restarted it, right?

Perhaps checking if the pid changes would be a more robust way to make sure this test isn't giving a false positive?